src/lib.rs

```diff
@@ -55,6 +55,54 @@ pub struct GUID {
     data: [u8; 16],
 }
 
+impl From<CGuid> for GUID {
+    fn from(item: CGuid) -> Self {
+        GUID::build_from_components(item.a, item.b, item.c, &item.d)
+    }
+}
+
+impl From<GUID> for CGuid {
+    fn from(item: GUID) -> Self {
+        CGuid {
+            a: item.data1(),
+            b: item.data2(),
+            c: item.data3(),
+            d: item.data4(),
+        }
+    }
+}
+
+impl fmt::Display for CGuid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{:08X}-{:04X}-{:04X}-{:04X}-{:08X}{:04X}",
+            self.a,
+            self.b,
+            self.c,
+            u16::from_be_bytes(self.d[0..2].try_into().unwrap()),
+            u32::from_be_bytes(self.d[2..6].try_into().unwrap()),
+            u16::from_be_bytes(self.d[6..8].try_into().unwrap()),
```

Comment:
I would prefer `.expect` with a note that the slice length is correct.

I wonder if the display impl could just be forwarded to the one on `GUID` and use a conversion from `CGuid`?

Reply:
> I would prefer .expect with a note that the slice length is correct.

Hmm, it can't really fail, the array is a fixed length array.

Could also do this:

```diff
@@ -398,9 +398,9 @@ impl fmt::Display for GUID {
             self.data1(),
             self.data2(),
             self.data3(),
-            u16::from_be_bytes(self.data[8..10].try_into().unwrap()),
-            u32::from_be_bytes(self.data[10..14].try_into().unwrap()),
-            u16::from_be_bytes(self.data[14..16].try_into().unwrap()),
+            u16::from_be_bytes([self.data[8], self.data[9]]),
+            u32::from_be_bytes([self.data[10], self.data[11], self.data[12], self.data[13]]),
+            u16::from_be_bytes([self.data[14], self.data[15]]),
         )
     }
 }
```

What do you think?

Reply:
> Hmm, it can't really fail, the array is a fixed length array.

Yeah I know, it is more a future proofing request. I avoid `.unwrap` as more of a 'prototype' call, whereas `.expect` is explicit that this should not fail.